magicsock: TestActiveDiscovery data race

```
==================
WARNING: DATA RACE
Write at 0x000001ab0458 by goroutine 120:
  tailscale.com/wgengine/magicsock.TestActiveDiscovery()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock_test.go:751 +0x96
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1306 +0x47

Previous read at 0x000001ab0458 by goroutine 123:
  tailscale.com/wgengine/magicsock.(*endpoint).sendPingsLocked()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3535 +0x38e
  tailscale.com/wgengine/magicsock.(*endpoint).heartbeat()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3345 +0x76d
  tailscale.com/wgengine/magicsock.(*endpoint).heartbeat-fm()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3315 +0x39

Goroutine 120 (running) created at:
  testing.(*T).Run()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:79 +0x22b

Goroutine 123 (finished) created at:
  time.goFunc()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/time/sleep.go:180 +0x49
==================
==================
WARNING: DATA RACE
Write at 0x000001ab0460 by goroutine 120:
  tailscale.com/wgengine/magicsock.TestActiveDiscovery.func2()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock_test.go:754 +0x3b
  tailscale.com/wgengine/magicsock.TestActiveDiscovery()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock_test.go:846 +0x149
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1306 +0x47

Previous read at 0x000001ab0460 by goroutine 131:
  tailscale.com/wgengine/magicsock.(*endpoint).startPingLocked()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3517 +0x444
  tailscale.com/wgengine/magicsock.(*endpoint).sendPingsLocked()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3546 +0x634
  tailscale.com/wgengine/magicsock.(*endpoint).heartbeat()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3345 +0x76d
  tailscale.com/wgengine/magicsock.(*endpoint).heartbeat-fm()
      D:/a/tailscale/tailscale/wgengine/magicsock/magicsock.go:3315 +0x39

Goroutine 120 (running) created at:
  testing.(*T).Run()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/testing/testing.go:1504 +0x9d1
  main.main()
      _testmain.go:79 +0x22b

Goroutine 131 (finished) created at:
  time.goFunc()
      C:/hostedtoolcache/windows/go/1.17.2/x64/src/time/sleep.go:180 +0x49
==================
--- FAIL: TestActiveDiscovery (2.15s)
```

--- a/wgengine/magicsock/magicsock_test.go
+++ b/wgengine/magicsock/magicsock_test.go
@@ -51,6 +51,13 @@ import (
 
 func init() {
 	os.Setenv("IN_TS_TEST", "1")
+
+	// Some of these tests lose a disco pong before establishing a
+	// direct connection, so instead of waiting 5 seconds in the
+	// test, reduce the wait period.
+	// (In particular, TestActiveDiscovery.)
+	discoPingInterval = 100 * time.Millisecond
+	pingTimeoutDuration = 100 * time.Millisecond
 }
 
 // WaitReady waits until the magicsock is entirely initialized and connected
@@ -742,18 +749,6 @@ func TestActiveDiscovery(t *testing.T) {
 		}
 		testActiveDiscovery(t, n)
 	})
-
-	// The following tests lose a disco pong before establishing a
-	// direct connection, so instead of waiting 5 seconds in the
-	// test, reduce the wait period.
-	origPingTimeoutDuration := pingTimeoutDuration
-	origDiscoPingInterval := discoPingInterval
-	discoPingInterval = 100 * time.Millisecond
-	pingTimeoutDuration = 100 * time.Millisecond
-	defer func() {
-		pingTimeoutDuration = origPingTimeoutDuration
-		discoPingInterval = origDiscoPingInterval
-	}()
 
 	t.Run("facing_easy_firewalls", func(t *testing.T) {
 		mstun := &natlab.Machine{Name: "stun"}

